### PR TITLE
LRU and LFU caches

### PR DESCRIPTION
**Pull request by vyark**

### CODE REVIEW

File: src/main/java/lru/LRUCache.java [hunk 1]
```diff
@@ -139,9 +130,9 @@ public void displayStatistics() {
     }
 
     private Long getAverageTime() {
```
Comment (fedorchuck) [inline src/main/java/lru/LRUCache.java:132]: DRY

File: src/main/java/shared/Item.java [hunk 2]
```diff
@@ -1,13 +1,4 @@
...
-
-@ToString
-public class Item {
-    private String key;
-
-    public Item(String key) {
-        this.key = key;
-    }
-
+public record Item(String key) {
```
Comment (fedorchuck) [inline src/main/java/shared/Item.java:3]: will it work?

**Review by fedorchuck — COMMENTED** [review]
could be merged if compiled

File: src/main/java/shared/Item.java (hunk 2 above)
Comment (vyark) [reply src/main/java/shared/Item.java:3]: yes, it works

**Pull request merged**